realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -594,11 +610,46 @@ public Builder schemaVersion(long schemaVersion) {
             return this;
         }
 
+        /**
+         * Update the schema version of the Realm and apply the provided migration.
+         * <p>
+         * The schema version must be equal to or higher than the schema version of the existing
+         * Realm file, if any, and the supplied {@link RealmMigration} must handled any schema changes.
+         * For <i>version only</i> updates the supplied
+         * {@link RealmMigration#migrate(DynamicRealm, long, long)} can be a <i>no operation</i>.
+         * <p>
+         * If the schema has changed without increasing the version or failing to resolve schema
+         * migration through the supplied {@link RealmMigration},, Realm will throw a
+         * {@link io.realm.exceptions.RealmMigrationNeededException}.
+         *
+         * @throws IllegalArgumentException If {@code schemaVersion} is negative or {@code migration} is null.
+         *
+         * @see #schemaVersion(long)
+         * @see #migration(RealmMigration)
+         */
+        public Builder schemaVersion(long schemaVersion, RealmMigration migration) {
+            if (schemaVersion < 0) {
+                throw new IllegalArgumentException("Realm schema version numbers must be 0 (zero) or higher. Yours was: " + schemaVersion);
+            }
+            if (migration == null) {
+                throw new IllegalArgumentException("Migration cannot be null");
+            }
+            this.schemaVersion = schemaVersion;
+            this.migration = migration;
+            return this;
+        }
+
         /**
          * Sets the {@link io.realm.RealmMigration} to be run if a migration is needed. If this migration fails to
          * upgrade the on-disc schema to the runtime schema, a {@link io.realm.exceptions.RealmMigrationNeededException}
          * will be thrown.
+         * <p>
+         * FIXME Also triggered on version update even though no schema change
+         * FIXME Not triggered if no version number is specified.
+         *
+         * @deprecated Use {@link #schemaVersion(long, RealmMigration)} instead.
          */
+        @Deprecated
```

Comment:
Normally this would be correct, but since is this being merged into a branch that will be released as a major (breaking) version. It is fine to just remove this outright. But we do need to add a line about in the CHANGELOG.md Breaking Changes section.

Reply:
👍